### DISCUSSION TITLE
ace editor, hide comments?

### DISCUSSION THREAD

**Discussion by ldijkman** [ajaxorg/ace#4854] [General]
Hello 

is it possible to hide / show comments in edited text
```
<!-- comment -->
// comment
/* comment */

```
greet luberth
https://ldijkman.github.io/Electra/

https://www.youtube.com/watch?v=5uapPPC8eQY

![2022-07-24-035113_1920x1080_scrot](https://user-images.githubusercontent.com/45427770/180628868-7d1e99a6-658a-4944-9399-8084c3a10aac.png)

**Comment by nightwing** [comment]
You can use fold all comments command https://github.com/ajaxorg/ace/blob/145328b96083c6b7c306d3dcf841948f73486919/lib/ace/commands/default_commands.js#L139-L146

**Reply by ldijkman** [reply]
thanks much
i will see if i can figure out how to implement it ;-)

**Reply by nightwing** [reply]
You simply need to call `editor.execCommand("foldAllComments")`

Only subtleness is that to work it needs the mode to already be loaded, so if it doesn't work at startup, you may use a callback of `setMode`
```js
editor.session.setMode("ace/mode/javascript",_=>editor.execCommand("foldAllComments")) 
```
or a listener for changeMode
```js
editor.session.on("changeMode", _=>editor.execCommand("foldAllComments")) 
```

**Comment by ldijkman** [comment]
it does something (green higlight line 48) 
but will have to play some more ;-

```
<input type="button" value="fold" onClick='ace.edit("editor").execCommand("foldAllComments")' title="fold comments">
```

![2022-07-24-135448_1920x1080_scrot](https://user-images.githubusercontent.com/45427770/180645833-8534074d-c5f8-4854-b916-258048c30da7.png)
)

**Comment by ldijkman** [comment]
looks like not all comments are folded in html

```
<!--  
I PREFER PLAIN JAVASCRIPT 
--> 

<!-- comment -->

<script>
// comment //

/* comment */

/*
 comment 
*/

<!--  
I PREFER PLAIN JAVASCRIPT 
--> 

<!-- comment -->

</script>

<!-- https://ldijkman.github.io/Electra/schedule.html -->
	
</script>
```
![2022-07-24-172256_1920x1080_scrot](https://user-images.githubusercontent.com/45427770/180654116-cd20c3dd-2fca-4cc9-8075-293a507df5d9.png)

**Reply by nightwing** [reply]
It folds multiline comments that are syntactically valid. AFAIK `<!--` comment is not allowed in script tag

**Reply by ldijkman** [reply]
yes you are right, thank you

saw also a ctrl+shift+/ command behaves different in html / css / script

use itt now for a paste option menu

```
function paste() {  
var myval = document.getElementById("myList").value;  
var htmltext;

if(myval==1){
 //    /* */ or <!-- -->
  ace.edit("editor").execCommand("toggleBlockComment");
  
  // htmltext="<!--     -->";
  // ace.edit("editor").session.insert(ace.edit("editor").getCursorPosition(), htmltext);
  // move cursor rom end of line  inside the comment part of paste
  // ace.edit('editor').gotoLine(ace.edit("editor").selection.getCursor().row+1,ace.edit("editor").selection.getCursor().column-6);
}
}

<select id = "myList" onchange = "paste()" style="width: 55px;" title="paste at cursor position"> 
<option value="0">paste</option> 
<option value="1"> comment /* */ or &lt;!-- --&gt; Ctrl+Shift+/&nbsp;</option>  
<option value="2"> html page </option>  
<option value="3"> img src </option>  
<option value="4"> a href </option>  
<option value="5"> script src= </option>  
<option value="6"> \&\nbsp; </option> 
<option value="7"> \&\emsp; </option> 
<option value="8"> p align right </option> 
<option value="9"> iframe </option> 
<option value="10"> meta refresh 10sec</option> 
</select>  
```
